src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2246,6 +2245,25 @@ def update_azure_service_mesh_profile(self) -> ServiceMeshProfile:
             new_profile.istio.certificate_authority.plugin.cert_chain_object_name = cert_chain_object_name
             updated = True
 
+        # deal with mesh upgrade commands
+        mesh_upgrade_command = self.raw_param.get("mesh_upgrade_command", None)
+        if mesh_upgrade_command is not None:
+            if len(new_profile.istio.revisions) < 2:
+                raise ArgumentUsageError('Azure Service Mesh upgrade is not in progress.')
+
+            if mesh_upgrade_command is not None and mesh_upgrade_command == "complete":
```

Comment:
isn't this part of the if statement duplicated/implicit? `if mesh_upgrade_command is not none`
Wouldn't have reached this line if line 2250 hadn't already been true

Reply:
This is correct. Will fix this as part of the subsequent PR.